src/test/modelconfig_test.cpp

```diff
@@ -221,6 +221,98 @@ TEST(ModelConfig, parseShapeParam) {
     EXPECT_EQ(status, ovms::StatusCode::SHAPE_WRONG_FORMAT);
 }
 
+TEST(ModelConfig, parseShapeDynamicParam) {
+    using namespace ovms;
+    ovms::ModelConfig config;
+    // Valid
+    std::string valid_str1 = "(64:128,128,256:512,300:301)";
+    std::string valid_str2 = "{\"input\": \"(1, 3:6, 3, 200:100000)\"}";
+    std::string valid_str3 = "{\"input\": \"auto\", \"extra_input\": \"(10:20)\"}";
+
+    ASSERT_EQ(config.parseShapeParameter(valid_str1), StatusCode::OK);
+    auto shapes = config.getShapes_2();
+    EXPECT_EQ(shapes[ovms::ANONYMOUS_INPUT_NAME].shapeMode, ovms::FIXED);
+    EXPECT_EQ(shapes[ovms::ANONYMOUS_INPUT_NAME].shape, (ovms::Shape{{64, 128}, 128, {256, 512}, {300, 301}}));
+
+    auto shape = ovms::Shape{1, 5, 10};
+
+    ASSERT_EQ(config.parseShapeParameter(valid_str2), StatusCode::OK);
```

Comment:
Mv valid_str2 to line just above for readability purposess. SImilarly iwth valid_str3

Reply:
The readibility is very subjective in this case, I mirrored the test for non dynamic shapes and it was more clear to me what is valid and what is invalid